File: Pruefungen/Winter17-18.java
// Lösung der Aufgabe "Freie nebeneinander liegende Kinositze auf dem Kreuzfahrtschiff"
//
// 7 Sitzreihen, 30 Plätze pro Reihe
// Sitzreihe 1: Sitznummern 101 bis 130
// Sitzreihe 2: Sitznummern 201 bis 230

public int freieSitze(int anzahlSitze) {
    for (int reihe = 0; reihe < 7; reihe++) {
        // Anzahl Sitze nebeneinander
        int anzahl = 0;
        for (int sitz = 0; sitz < 30; sitz++) {
            if (kino[reihe][sitz]) {
                // Einen freien Sitz gefunden
                anzahl++;
                if (anzahl == anzahlSitze) {
                    // Die benötigte Anzahl wurde erreicht
                    return (reihe + 1) * 100 + anzahl + 1;
                }
            } else {
                // Kein freier Sitzplatz: zurück zum Start
                anzahl = 0;
            }
        }
    }
    // Nix gefunden
    return 0;
}

File: Pruefungen/Winter17-18.py
# Lösung der Aufgabe "Freie Kinositze auf dem Kreuzfahrtschiff"

# 7 Sitzreihen, 30 Plätze pro Reihe
# Sitzreihe 1: Sitznummern 101 bis 130
# Sitzreihe 2: Sitznummern 201 bis 230

def freie_sitze(anzahl_sitze):
    for reihe in range(7):
        # Anzahl Sitze nebeneinander
        anzahl = 0
        for sitz in range(30):
            if kino[reihe][sitz]:
                # Einen freien Sitz gefunden
                anzahl = anzahl + 1
                if anzahl == anzahl_sitze:
                    # Die benötigte Anzahl wurde erreicht
                    return (reihe + 1) * 100 + anzahl + 1
            else:
                # Kein freier Sitzplatz: zurück zum Start
                anzahl = 0
    # Nix gefunden
    return 0
